be/Controller/UsersCtrl.php

```diff
@@ -99,56 +99,61 @@ public function addUser(Request $request, Response $response): Response
         return $response->withJson([]);
     }
 
-    private function getGame(
-        User $winnerUser,
-        User $looserUser,
-        int $oldWinnerRating,
-        int $oldLooserRating,
-        int $ratingDiff
-    ): Game {
-        $game = new Game();
-        $game->setWinnerUser($winnerUser);
-        $game->setLooserUser($looserUser);
-        $game->setWinnerRatingBefore($oldWinnerRating);
-        $game->setLooserRatingBefore($oldLooserRating);
-        $game->setRatingDiff($ratingDiff);
+    public function updateRatings(
+        Request $request,
+        Response $response
+    ): Response {
+        $json = $request->getBody();
+        $usersCodes = json_decode($json, true);
+        $winnerUserCode = $usersCodes['winnerUserCode'];
+        $looserUserCode = $usersCodes['looserUserCode'];
 
-        return $game;
-    }
+        $this->valdateUpdateRatingsCodes($winnerUserCode, $looserUserCode);
 
-    private function calcNewRatings(
-        int $oldWinnerRating,
-        int $oldLooserRating
-    ): array {
-        $kfactor = 32;
-
-        $winnerLooserDiff = $oldLooserRating - $oldWinnerRating;
-        $pWinner = 1 / (1 + (10 ** ($winnerLooserDiff / 400))); // propability of winning winner user
+        /** @var User $winnerUser */
+        $winnerUser = $this->userRepository->findOneBy(['code' => $winnerUserCode]);
+        /** @var User $looserUser */
+        $looserUser = $this->userRepository->findOneBy(['code' => $looserUserCode]);
 
-        $ratingDiff = round($kfactor * (1 - $pWinner));
-        $newWinnerRating = $oldWinnerRating + $ratingDiff;
-        $newLooserRating = $oldLooserRating - $ratingDiff;
+        $warningMsg = $this->getUpdateUserWarningMsg($winnerUser, $looserUser);
+        if ($warningMsg) {
+            return $response->withJson(['status' => 'warning', 'warningMsg' => $warningMsg]);
+        }
 
-        return [$newWinnerRating, $newLooserRating];
+        $ratingDiff = $this->updateRatingsInDb($winnerUser, $looserUser);
+        return $response->withJson(['status' => 'success', 'ratingDiff' => $ratingDiff]);
     }
 
-    private function updateRatingsInDb(
-        int $winnerUserNid,
-        int $looserUserNid
-    ): int {
-        /** @var User $winnerUser */
-        $winnerUser = $this->userRepository->find($winnerUserNid);
-        /** @var User $looserUser */
-        $looserUser = $this->userRepository->find($looserUserNid);
-
-        if (!isset($winnerUser, $looserUser)) {
-            throw new \InvalidArgumentException('Winner or looser does not exist');
+    private function valdateUpdateRatingsCodes(string $winnerUserCode, string $looserUserCode): void
```

Comment:
Co do tej funkcji to nie musisz na nią zwracać szczególną uwagę bo w pull requeście cz.2 ją wywalam